core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -481,13 +481,47 @@ trait EntityComponent {
           }
 
           //actually build the records we're going to send to the db
-          val insertRecs = attributeMapToRecs(insertAttrs)
-          val updateRecs = attributeMapToRecs(updateAttrs)
-          val deleteIds = (for {
+          var insertRecs = attributeMapToRecs(insertAttrs)
+          var updateRecs = attributeMapToRecs(updateAttrs)
+          var deleteIds = (for {
             attributeName <- deletes
-          } yield existingAttrsToRecordIds.get(attributeName)).flatten
+          } yield existingAttrsToRecordIds.get(attributeName)).flatten.flatten
 
-          entityAttributeQuery.patchAttributesAction(insertRecs, updateRecs, deleteIds.flatten, entityAttributeScratchQuery.insertScratchAttributes)
+
+          def checkAndUpdateRecsForListAttr(existingAttrListSize: Int, updateAttrListSize: Int, updateAttr: AttributeName): Unit = {
+            if (updateAttrListSize > existingAttrListSize) {
+              // since the size of the list has increased, move these new records to insertRecs
+              val (newInsertRecs, newUpdateRecs) = updateRecs.partition {x =>
+                updateAttr.name.equals(x.name) && updateAttr.namespace.equals(x.namespace) && (x.listIndex.get > (existingAttrListSize - 1))
+              }
+
+              insertRecs = insertRecs ++ newInsertRecs
+              updateRecs = newUpdateRecs
+            } else if (updateAttrListSize < existingAttrListSize) {
+              // since the size of the list has decreased, delete the extra rows from table
+              deleteIds = existingAttrsToRecordIds(updateAttr).takeRight(existingAttrListSize - updateAttrListSize) ++ deleteIds
+            }
+          }
+
+          /*
+            Additional check for entities whose Attribute value type is AttributeValueList in response to bug mentioned in WA-32
+            (https://broadworkbench.atlassian.net/browse/WA-32). Currently the update query is such that it matches the
+            listIndex of each attribute value and updates it. Hence if the size of the list changes, those changes are
+            not reflected in the table resulting in the behavior mentioned in the ticket.
+            Hence, for any attribute in the update list with value type as AttributeValueList, check if the size of the
+            list has changed. If yes, based on increase or decrease of the list size, add those extra records into insertRecs
+            or delete extra records from the entity table respectively.
+           */
+          updateAttrs.foreach {
+            case (attr, list: AttributeValueList) => {
```

Comment:
Thank you for the feedback Doug. I will modify the code to avoid mutable state. I will also look at `AttributeEntityReferenceList`. While I was trying to fix the bug I was focused on `AttributeValueList` and didn't take a look at `AttributeEntityReferenceList`.